packages/core/src/delegation/DelegationNode.ts

```diff
@@ -98,52 +239,83 @@ export default class DelegationNode extends DelegationBaseNode
+    if (this.isRoot()) {
+      return storeAsRoot(this)
+      // eslint-disable-next-line no-else-return
+    } else {
+      if (!signature) {
+        throw SDKErrors.ERROR_DELEGATION_SIGNATURE_MISSING
+      }
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return storeAsDelegation(this, signature!)
     }
-    return rootNode
+  }
+
+  isRoot(): boolean {
+    return this.id === this.hierarchyId && !this.parentId
   }
 
   /**
-   * [ASYNC] Fetches the parent node of this delegation node.
+   * [ASYNC] Verifies the delegation node by querying it from chain and checking its revocation status.
    *
-   * @returns Promise containing the parent as [[DelegationBaseNode]] or [null].
+   * @returns Promise containing a boolean flag.
    */
-
-  public async getParent(): Promise<DelegationBaseNode | null> {
-    if (!this.parentId || this.parentId === this.rootId) {
-      // parent must be root
-      return this.getRoot()
-    }
-    return query(this.parentId)
+  public async verify(): Promise<boolean> {
+    const node = await query(this.id)
+    return node !== null && !node.revoked
   }
 
   /**
-   * [ASYNC] Stores the delegation node on chain.
+   * [ASYNC] Checks on chain whether a identity with the given address is delegating to the current node.
    *
-   * @param signature Signature of the delegate to ensure it is done under the delegate's permission.
-   * @returns Promise containing a unsigned SubmittableExtrinsic.
+   * @param address The address of the identity.
+   *
+   * @returns An object containing a `node` owned by the identity if it is delegating, plus the number of `steps` traversed. `steps` is 0 if the address is owner of the current node.
    */
-  public async store(signature: string): Promise<SubmittableExtrinsic> {
-    log.info(`:: store(${this.id})`)
-    return store(this, signature)
+  public async findAncestorOwnedBy(
+    address: Identity['address']
+  ): Promise<{ steps: number; node: DelegationNode | null }> {
+    if (this.account === address) {
+      return {
+        steps: 0,
+        node: this,
+      }
+    }
+    const parent = await this.getParent()
+    if (parent) {
+      const result = await parent.findAncestorOwnedBy(address)
+      result.steps += 1
+      return result
+    }
+    return {
+      steps: 0,
+      node: null,
+    }
   }
 
   /**
-   * [ASYNC] Verifies the delegation node by querying it from chain and checking its revocation status.
+   * [ASYNC] Recursively counts all nodes that descend from the current node (excluding the current node).
    *
-   * @returns Promise containing a boolean flag.
+   * @returns Promise resolving to the node count.
    */
-  public async verify(): Promise<boolean> {
-    const node = await query(this.id)
-    return node !== null && !node.revoked
+  public async subtreeNodeCount(): Promise<number> {
+    const children = await this.getChildren()
```

Comment:
what do you think about refetching here to make sure we are up to date?

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/406/commits/8a20ab281b2ba7d57cd26af3a0736060d6fa3309.